sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProperties.java

```diff
@@ -0,0 +1,189 @@
+        this.enabled = enabled;
+    }
+
+    public String getUri() {
+        return uri;
+    }
+
+    public void setUri(String uri) {
+        this.uri = uri;
+    }
+
+    public Long getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    public void setRefreshInterval(Long refreshInterval) {
+        this.refreshInterval = refreshInterval;
+    }
+
+    public List<String> getSecretKeys() {
+        return secretKeys;
+    }
+
+    public void setSecretKeys(List<String> secretKeys) {
+        this.secretKeys = secretKeys;
+    }
+
+    public String getOrder() {
+        return order;
+    }
+
+    public void setOrder(String order) {
+        this.order = order;
+    }
+
+    public String getCaseSensitiveKeys() {
+        return caseSensitiveKeys;
+    }
+
+    public void setCaseSensitiveKeys(String caseSensitiveKeys) {
+        this.caseSensitiveKeys = caseSensitiveKeys;
+    }
+
+    public String getAllowTelemetry() {
+        return allowTelemetry;
+    }
+
+    public void setAllowTelemetry(String allowTelemetry) {
+        this.allowTelemetry = allowTelemetry;
+    }
+
+    private Boolean enabled;
+    private List<String> secretKeys;
+    private Long refreshInterval = Constants.DEFAULT_REFRESH_INTERVAL_MS;
+    private String allowTelemetry;
+    /**
+     * Defines the constant for the property that enables/disables case sensitive keys.
+     */
+    private String caseSensitiveKeys;
+    private String certificatePassword;
+    private String certificatePath;
+    private String clientId;
+    private String clientKey;
+    /**
+     * The constant used to define the order of the key vaults you are
+     * delivering (comma delimited, e.g 'my-vault, my-vault-2').
+     */
+    private String order;
+    private String tenantId;
+    private String uri;
+
+
+    public enum Property {
+        ALLOW_TELEMETRY("allow-telemetry"),
+        CASE_SENSITIVE_KEYS("case-sensitive-keys"),
+        CERTIFICATE_PASSWORD("certificate-password"),
+        CERTIFICATE_PATH("certificate-path"),
+        CLIENT_ID("client-id"),
+        CLIENT_KEY("client-key"),
+        ENABLED("enabled"),
+        ORDER("order"),
+        REFRESH_INTERVAL("refresh-interval"),
+        SECRET_KEYS("secret-keys"),
+        TENANT_ID("tenant-id"),
+        URI("uri");
+
+        private final String name;
+
+        String getName() {
+            return name;
+        }
+
+        Property(String name) {
+            this.name = name;
+        }
+    }
+
+    public static String getPropertyName(Property property) {
+        return String.join(DELIMITER, PREFIX, property.getName());
+    }
+
+    public static String getPropertyName(String normalizedName, Property property) {
+        return Stream.of(PREFIX, normalizedName, property.getName())
+            .map(String::trim)
+            .filter(s -> !s.isEmpty())
+            .collect(Collectors.joining(DELIMITER));
+    }
+
+    public static void main(String[] args) {
```

Comment:
Should we remove this `main` method?

Reply:
If I removed `main` method, `mvn clean install` will report `Unused field Error.`

Reply:
It should be removed - we don't want users to run this so I can't see why we would include it.

Reply:
Already remoted. Thank you for your check. 🙇